### ISSUE TITLE
xdp-tools-1.4.2 bump causes build problems on qoriq

### ISSUE THREAD

**Issue by stintel** [openwrt/openwrt#16320]
[compile.txt](https://github.com/user-attachments/files/17356149/compile.txt)
### Describe the bug

The xdp-tools-1.4.2 bump causes build problems on qoriq:
[compile.txt](https://github.com/user-attachments/files/17356177/compile.txt)

Reverting f3080677f506d2670c1911a91c19eec4ace5d085 solves the problem.

@rockdrilla 

### OpenWrt version

r27297-2bec6f48e6

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

qoriq/generic

### Device

WatchGuard Firebox M300

### Image kind

Self-built image

### Diffconfig

```text
CONFIG_TARGET_qoriq=y
CONFIG_TARGET_qoriq_generic=y
CONFIG_TARGET_qoriq_generic_DEVICE_watchguard_firebox-m300=y
CONFIG_DEVEL=y
CONFIG_BIND_ENABLE_DOH=y
CONFIG_BPF_TOOLCHAIN_HOST=y
CONFIG_BPF_TOOLCHAIN_HOST_PATH=""
# CONFIG_BPF_TOOLCHAIN_NONE is not set
CONFIG_BUILD_LOG=y
CONFIG_BUSYBOX_DEFAULT_PIE=y
CONFIG_COLLECT_KERNEL_DEBUG=y
CONFIG_DEBUG=y
CONFIG_DROPBEAR_MODERN_ONLY=y
CONFIG_DROPBEAR_ZLIB=y
CONFIG_DWARVES=y
# CONFIG_FEED_ecnmarker is not set
# CONFIG_FEED_floattest is not set
# CONFIG_FEED_matter is not set
# CONFIG_FEED_usteer_web_ucode is not set
# CONFIG_FEED_video is not set
CONFIG_GNUTLS_ALPN=y
CONFIG_GNUTLS_ANON=y
CONFIG_GNUTLS_DTLS_SRTP=y
CONFIG_GNUTLS_HEARTBEAT=y
CONFIG_GNUTLS_OCSP=y
CONFIG_GNUTLS_PSK=y
CONFIG_HAS_BPF_TOOLCHAIN=y
CONFIG_IPERF_ENABLE_MULTICAST=y
# CONFIG_JSON_OVERVIEW_IMAGE_INFO is not set
CONFIG_KEEPALIVED_BFD=y
CONFIG_KEEPALIVED_CHECKSUM_COMPAT=y
CONFIG_KEEPALIVED_FWMARK=y
CONFIG_KEEPALIVED_LINKBEAT=y
CONFIG_KEEPALIVED_LVS=y
CONFIG_KEEPALIVED_LVS_64BIT_STATS=y
CONFIG_KEEPALIVED_LVS_SYNCD=y
CONFIG_KEEPALIVED_NFTABLES=y
CONFIG_KEEPALIVED_ROUTES=y
CONFIG_KEEPALIVED_SHA1=y
CONFIG_KEEPALIVED_SNMP_CHECKER=y
CONFIG_KEEPALIVED_SNMP_REPLY_V3_FOR_V2=y
CONFIG_KEEPALIVED_SNMP_RFC3=y
CONFIG_KEEPALIVED_VRRP=y
CONFIG_KEEPALIVED_VRRP_AUTH=y
CONFIG_KERNEL_AUDIT=y
# CONFIG_KERNEL_BLK_DEV_THROTTLING is not set
CONFIG_KERNEL_BPF_EVENTS=y
CONFIG_KERNEL_CC_STACKPROTECTOR_NONE=y
# CONFIG_KERNEL_CC_STACKPROTECTOR_REGULAR is not set
# CONFIG_KERNEL_CFS_BANDWIDTH is not set
CONFIG_KERNEL_CGROUP_DEVICE=y
CONFIG_KERNEL_CGROUP_FREEZER=y
CONFIG_KERNEL_DEBUG_INFO_BTF=y
CONFIG_KERNEL_DEBUG_INFO_BTF_MODULES=y
# CONFIG_KERNEL_DEBUG_INFO_REDUCED is not set
CONFIG_KERNEL_FTRACE=y
CONFIG_KERNEL_IO_URING=y
# CONFIG_KERNEL_KEYS is not set
CONFIG_KERNEL_KPROBES=y
CONFIG_KERNEL_KPROBE_EVENTS=y
# CONFIG_KERNEL_MEMCG_SWAP is not set
CONFIG_KERNEL_NET_CLS_CGROUP=y
CONFIG_KERNEL_PERF_EVENTS=y
CONFIG_KERNEL_SLABINFO=y
CONFIG_KERNEL_SLUB_DEBUG=y
CONFIG_KERNEL_SLUB_DEBUG_ON=y
# CONFIG_KERNEL_STACKPROTECTOR is not set
# CONFIG_KERNEL_WERROR is not set
CONFIG_KERNEL_XDP_SOCKETS=y
CONFIG_LIBSSH2_OPENSSL=y
CONFIG_LINUX_6_6=y
CONFIG_LLDPD_WITH_CUSTOM=y
CONFIG_LLDPD_WITH_DOT1=y
CONFIG_LLDPD_WITH_DOT3=y
CONFIG_LLDPD_WITH_LLDPMED=y
CONFIG_LLDPD_WITH_PRIVSEP=y
CONFIG_LLDPD_WITH_SNMP=y
# CONFIG_LUA_ECO_MBEDTLS is not set
CONFIG_LUA_ECO_OPENSSL=y
CONFIG_MOSQUITTO_CTRL=y
CONFIG_MOSQUITTO_DYNAMIC_SECURITY=y
CONFIG_MOSQUITTO_LWS=y
CONFIG_MOSQUITTO_PASSWD=y
CONFIG_OPENSSL_ENGINE=y
CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
CONFIG_OPENSSL_WITH_ASM=y
CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
CONFIG_OPENSSL_WITH_CMS=y
CONFIG_OPENSSL_WITH_DEPRECATED=y
CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_SRP=y
CONFIG_OPENSSL_WITH_TLS13=y
CONFIG_PACKAGE_6in4=y
CONFIG_PACKAGE_attendedsysupgrade-common=y
CONFIG_PACKAGE_audit-utils=y
CONFIG_PACKAGE_bind-client=y
CONFIG_PACKAGE_bind-libs=y
CONFIG_PACKAGE_bird2=y
CONFIG_PACKAGE_bird2c=y
CONFIG_PACKAGE_bpftool-full=y
CONFIG_PACKAGE_cJSON=y
CONFIG_PACKAGE_cgi-io=y
CONFIG_PACKAGE_chrony-nts=y
CONFIG_PACKAGE_conntrack=y
CONFIG_PACKAGE_conntrackd=y
CONFIG_PACKAGE_ddns-scripts=y
CONFIG_PACKAGE_ddns-scripts-nsupdate=y
CONFIG_PACKAGE_ddns-scripts-services=y
# CONFIG_PACKAGE_dnsmasq is not set
CONFIG_PACKAGE_dnsmasq-full=y
CONFIG_PACKAGE_dnsmasq_full_auth=y
CONFIG_PACKAGE_dnsmasq_full_conntrack=y
CONFIG_PACKAGE_dnsmasq_full_dhcp=y
CONFIG_PACKAGE_dnsmasq_full_dhcpv6=y
CONFIG_PACKAGE_dnsmasq_full_dnssec=y
CONFIG_PACKAGE_dnsmasq_full_nftset=y
CONFIG_PACKAGE_dnsmasq_full_noid=y
CONFIG_PACKAGE_dnsmasq_full_tftp=y
CONFIG_PACKAGE_drill=y
# CONFIG_PACKAGE_e2fsprogs is not set
CONFIG_PACKAGE_etherwake=y
CONFIG_PACKAGE_ethtool-full=y
CONFIG_PACKAGE_f2fsck=y
CONFIG_PACKAGE_fstrim=y
CONFIG_PACKAGE_gdb=y
CONFIG_PACKAGE_gdbserver=y
CONFIG_PACKAGE_glib2=y
CONFIG_PACKAGE_iftop=y
CONFIG_PACKAGE_ip-bridge=y
CONFIG_PACKAGE_ip-full=y
CONFIG_PACKAGE_iperf=y
CONFIG_PACKAGE_iperf3=y
CONFIG_PACKAGE_ipset=y
CONFIG_PACKAGE_iputils-tracepath=y
CONFIG_PACKAGE_keepalived=y
CONFIG_PACKAGE_kmod-crypto-acompress=y
CONFIG_PACKAGE_kmod-crypto-aead=y
CONFIG_PACKAGE_kmod-crypto-authenc=y
CONFIG_PACKAGE_kmod-crypto-cbc=y
CONFIG_PACKAGE_kmod-crypto-ccm=y
CONFIG_PACKAGE_kmod-crypto-chacha20poly1305=y
CONFIG_PACKAGE_kmod-crypto-ctr=y
CONFIG_PACKAGE_kmod-crypto-deflate=y
CONFIG_PACKAGE_kmod-crypto-des=y
CONFIG_PACKAGE_kmod-crypto-ecb=y
CONFIG_PACKAGE_kmod-crypto-echainiv=y
CONFIG_PACKAGE_kmod-crypto-gcm=y
CONFIG_PACKAGE_kmod-crypto-geniv=y
CONFIG_PACKAGE_kmod-crypto-gf128=y
CONFIG_PACKAGE_kmod-crypto-ghash=y
CONFIG_PACKAGE_kmod-crypto-hmac=y
CONFIG_PACKAGE_kmod-crypto-lib-chacha20=y
CONFIG_PACKAGE_kmod-crypto-lib-chacha20poly1305=y
CONFIG_PACKAGE_kmod-crypto-lib-curve25519=y
CONFIG_PACKAGE_kmod-crypto-lib-poly1305=y
CONFIG_PACKAGE_kmod-crypto-manager=y
CONFIG_PACKAGE_kmod-crypto-md5=y
CONFIG_PACKAGE_kmod-crypto-null=y
CONFIG_PACKAGE_kmod-crypto-rng=y
CONFIG_PACKAGE_kmod-crypto-seqiv=y
CONFIG_PACKAGE_kmod-crypto-sha1=y
CONFIG_PACKAGE_kmod-crypto-sha256=y
CONFIG_PACKAGE_kmod-crypto-sha3=y
CONFIG_PACKAGE_kmod-crypto-sha512=y
CONFIG_PACKAGE_kmod-crypto-user=y
# CONFIG_PACKAGE_kmod-gpio-button-hotplug is not set
CONFIG_PACKAGE_kmod-ifb=y
CONFIG_PACKAGE_kmod-inet-diag=y
CONFIG_PACKAGE_kmod-input-core=y
CONFIG_PACKAGE_kmod-ipsec=y
CONFIG_PACKAGE_kmod-ipsec4=y
CONFIG_PACKAGE_kmod-ipsec6=y
CONFIG_PACKAGE_kmod-ipt-conntrack=y
CONFIG_PACKAGE_kmod-ipt-core=y
CONFIG_PACKAGE_kmod-ipt-ipset=y
CONFIG_PACKAGE_kmod-iptunnel=y
CONFIG_PACKAGE_kmod-iptunnel4=y
CONFIG_PACKAGE_kmod-iptunnel6=y
CONFIG_PACKAGE_kmod-l2tp=y
CONFIG_PACKAGE_kmod-lib-textsearch=y
CONFIG_PACKAGE_kmod-lib-zlib-deflate=y
CONFIG_PACKAGE_kmod-lib-zlib-inflate=y
CONFIG_PACKAGE_kmod-macvlan=y
CONFIG_PACKAGE_kmod-mdio-netlink=y
CONFIG_PACKAGE_kmod-mii=y
CONFIG_PACKAGE_kmod-netlink-diag=y
CONFIG_PACKAGE_kmod-nf-conntrack-netlink=y
CONFIG_PACKAGE_kmod-nf-ipt=y
CONFIG_PACKAGE_kmod-nf-ipt6=y
CONFIG_PACKAGE_kmod-nf-ipvs=y
CONFIG_PACKAGE_kmod-nf-nat6=y
CONFIG_PACKAGE_kmod-nf-nathelper=y
CONFIG_PACKAGE_kmod-nf-nathelper-extra=y
CONFIG_PACKAGE_kmod-nfnetlink-queue=y
CONFIG_PACKAGE_kmod-nft-bridge=y
CONFIG_PACKAGE_kmod-nft-netdev=y
CONFIG_PACKAGE_kmod-nls-base=y
CONFIG_PACKAGE_kmod-pppol2tp=y
CONFIG_PACKAGE_kmod-pstore=y
CONFIG_PACKAGE_kmod-ramoops=y
CONFIG_PACKAGE_kmod-reed-solomon=y
CONFIG_PACKAGE_kmod-sched=y
CONFIG_PACKAGE_kmod-sched-act-vlan=y
CONFIG_PACKAGE_kmod-sched-bpf=y
CONFIG_PACKAGE_kmod-sched-cake=y
CONFIG_PACKAGE_kmod-sched-core=y
CONFIG_PACKAGE_kmod-sched-flower=y
CONFIG_PACKAGE_kmod-sched-fq-pie=y
CONFIG_PACKAGE_kmod-sched-mqprio=y
CONFIG_PACKAGE_kmod-sched-mqprio-common=y
CONFIG_PACKAGE_kmod-sched-pie=y
CONFIG_PACKAGE_kmod-sctp=y
CONFIG_PACKAGE_kmod-sit=y
CONFIG_PACKAGE_kmod-tcp-bbr=y
CONFIG_PACKAGE_kmod-tls=y
CONFIG_PACKAGE_kmod-udptunnel4=y
CONFIG_PACKAGE_kmod-udptunnel6=y
CONFIG_PACKAGE_kmod-usb-core=y
CONFIG_PACKAGE_kmod-usb-ehci=y
CONFIG_PACKAGE_kmod-usb-net=y
CONFIG_PACKAGE_kmod-usb-net-cdc-ether=y
CONFIG_PACKAGE_kmod-usb-net-cdc-mbim=y
CONFIG_PACKAGE_kmod-usb-net-cdc-ncm=y
CONFIG_PACKAGE_kmod-usb-serial=y
CONFIG_PACKAGE_kmod-usb-serial-cp210x=y
CONFIG_PACKAGE_kmod-usb-serial-ftdi=y
CONFIG_PACKAGE_kmod-usb-serial-pl2303=y
CONFIG_PACKAGE_kmod-usb-wdm=y
CONFIG_PACKAGE_kmod-usb2=y
CONFIG_PACKAGE_kmod-veth=y
CONFIG_PACKAGE_kmod-vxlan=y
CONFIG_PACKAGE_kmod-wireguard=y
CONFIG_PACKAGE_kmod-xdp-sockets-diag=y
CONFIG_PACKAGE_kmod-xfrm-interface=y
# CONFIG_PACKAGE_knot-resolver_dnstap is not set
CONFIG_PACKAGE_libatomic=y
CONFIG_PACKAGE_libattr=y
CONFIG_PACKAGE_libaudit=y
CONFIG_PACKAGE_libauparse=y
CONFIG_PACKAGE_libbfd=y
CONFIG_PACKAGE_libbpf=y
CONFIG_PACKAGE_libbz2=y
CONFIG_PACKAGE_libcap=y
CONFIG_PACKAGE_libcap-ng=y
CONFIG_PACKAGE_libcares=y
# CONFIG_PACKAGE_libcomerr is not set
CONFIG_PACKAGE_libctf=y
CONFIG_PACKAGE_libdw=y
# CONFIG_PACKAGE_libe2p is not set
CONFIG_PACKAGE_libelf=y
CONFIG_PACKAGE_libev=y
CONFIG_PACKAGE_libevdev=y
CONFIG_PACKAGE_libevent2=y
# CONFIG_PACKAGE_libext2fs is not set
CONFIG_PACKAGE_libffi=y
CONFIG_PACKAGE_libgmp=y
CONFIG_PACKAGE_libgnutls=y
CONFIG_PACKAGE_libiperf3=y
CONFIG_PACKAGE_libipset=y
CONFIG_PACKAGE_libiwinfo=y
CONFIG_PACKAGE_libiwinfo-data=y
CONFIG_PACKAGE_libiwinfo-lua=y
CONFIG_PACKAGE_libkmod=y
CONFIG_PACKAGE_libldns=y
CONFIG_PACKAGE_liblua=y
CONFIG_PACKAGE_liblua5.3=y
CONFIG_PACKAGE_liblucihttp=y
CONFIG_PACKAGE_liblucihttp-lua=y
CONFIG_PACKAGE_liblucihttp-ucode=y
CONFIG_PACKAGE_liblzma=y
CONFIG_PACKAGE_libmagic=y
# CONFIG_PACKAGE_libmbedtls is not set
CONFIG_PACKAGE_libmbim=y
CONFIG_PACKAGE_libmosquitto-ssl=y
CONFIG_PACKAGE_libmount=y
CONFIG_PACKAGE_libmpfr=y
CONFIG_PACKAGE_libncurses=y
CONFIG_PACKAGE_libnetfilter-conntrack=y
CONFIG_PACKAGE_libnetfilter-cthelper=y
CONFIG_PACKAGE_libnetfilter-cttimeout=y
CONFIG_PACKAGE_libnetfilter-queue=y
CONFIG_PACKAGE_libnetsnmp=y
CONFIG_PACKAGE_libnettle=y
CONFIG_PACKAGE_libnfnetlink=y
CONFIG_PACKAGE_libnghttp2=y
CONFIG_PACKAGE_libnl-core=y
CONFIG_PACKAGE_libnl-genl=y
CONFIG_PACKAGE_libnl-route=y
CONFIG_PACKAGE_libopcodes=y
CONFIG_PACKAGE_libopenssl=y
CONFIG_PACKAGE_libopenssl-conf=y
CONFIG_PACKAGE_libpcap=y
CONFIG_PACKAGE_libpci=y
CONFIG_PACKAGE_libpcre2=y
CONFIG_PACKAGE_libreadline=y
CONFIG_PACKAGE_libsensors=y
CONFIG_PACKAGE_libsmartcols=y
# CONFIG_PACKAGE_libss is not set
CONFIG_PACKAGE_libssh2=y
CONFIG_PACKAGE_libstdcpp=y
CONFIG_PACKAGE_libsysfs=y
CONFIG_PACKAGE_libtirpc=y
CONFIG_PACKAGE_libtraceevent=y
CONFIG_PACKAGE_libubus-lua=y
CONFIG_PACKAGE_libuci-lua=y
CONFIG_PACKAGE_libudev-zero=y
CONFIG_PACKAGE_liburcu=y
CONFIG_PACKAGE_libusb-1.0=y
# CONFIG_PACKAGE_libustream-mbedtls is not set
CONFIG_PACKAGE_libustream-openssl=y
CONFIG_PACKAGE_libuv=y
CONFIG_PACKAGE_libwebsockets-openssl=y
CONFIG_PACKAGE_libxdp=y
CONFIG_PACKAGE_libyaml=y
CONFIG_PACKAGE_linuxptp=y
CONFIG_PACKAGE_lldpd=y
CONFIG_PACKAGE_lm-sensors=y
CONFIG_PACKAGE_lscpu=y
CONFIG_PACKAGE_lsns=y
CONFIG_PACKAGE_lsof=y
CONFIG_PACKAGE_lua=y
CONFIG_PACKAGE_luasocket=y
CONFIG_PACKAGE_luci=y
CONFIG_PACKAGE_luci-app-attendedsysupgrade=y
CONFIG_PACKAGE_luci-app-ddns=y
CONFIG_PACKAGE_luci-app-firewall=y
CONFIG_PACKAGE_luci-app-opkg=y
CONFIG_PACKAGE_luci-app-upnp=y
CONFIG_PACKAGE_luci-app-wol=y
CONFIG_PACKAGE_luci-base=y
CONFIG_PACKAGE_luci-compat=y
CONFIG_PACKAGE_luci-lib-base=y
CONFIG_PACKAGE_luci-lib-ip=y
CONFIG_PACKAGE_luci-lib-ipkg=y
CONFIG_PACKAGE_luci-lib-json=y
CONFIG_PACKAGE_luci-lib-jsonc=y
CONFIG_PACKAGE_luci-lib-nixio=y
CONFIG_PACKAGE_luci-lib-uqr=y
CONFIG_PACKAGE_luci-light=y
CONFIG_PACKAGE_luci-lua-runtime=y
CONFIG_PACKAGE_luci-mod-admin-full=y
CONFIG_PACKAGE_luci-mod-network=y
CONFIG_PACKAGE_luci-mod-rpc=y
CONFIG_PACKAGE_luci-mod-status=y
CONFIG_PACKAGE_luci-mod-system=y
CONFIG_PACKAGE_luci-proto-ipv6=y
CONFIG_PACKAGE_luci-proto-ppp=y
CONFIG_PACKAGE_luci-proto-wireguard=y
CONFIG_PACKAGE_luci-ssl-openssl=y
CONFIG_PACKAGE_luci-theme-bootstrap=y
CONFIG_PACKAGE_mbim-utils=y
CONFIG_PACKAGE_mdio-tools=y
CONFIG_PACKAGE_mdnsd=y
CONFIG_PACKAGE_miniupnpd-nftables=y
CONFIG_PACKAGE_mosquitto-ssl=y
CONFIG_PACKAGE_musl-fts=y
CONFIG_PACKAGE_objdump=y
# CONFIG_PACKAGE_odhcpd-ipv6only is not set
CONFIG_PACKAGE_openssl-util=y
CONFIG_PACKAGE_pciids=y
CONFIG_PACKAGE_pciutils=y
CONFIG_PACKAGE_perf=y
CONFIG_PACKAGE_pkg-config=y
CONFIG_PACKAGE_ppp-mod-pppol2tp=y
CONFIG_PACKAGE_procps-ng=y
CONFIG_PACKAGE_procps-ng-ps=y
CONFIG_PACKAGE_prometheus-node-exporter-ucode=y
CONFIG_PACKAGE_prometheus-node-exporter-ucode-netstat=y
CONFIG_PACKAGE_prometheus-node-exporter-ucode-openwrt=y
CONFIG_PACKAGE_prometheus-node-exporter-ucode-snmp6=y
CONFIG_PACKAGE_qosify=y
CONFIG_PACKAGE_resolveip=y
CONFIG_PACKAGE_ripgrep=y
CONFIG_PACKAGE_rng-tools=y
CONFIG_PACKAGE_rpcd=y
CONFIG_PACKAGE_rpcd-mod-file=y
CONFIG_PACKAGE_rpcd-mod-iwinfo=y
CONFIG_PACKAGE_rpcd-mod-luci=y
CONFIG_PACKAGE_rpcd-mod-rpcsys=y
CONFIG_PACKAGE_rpcd-mod-rrdns=y
CONFIG_PACKAGE_rpcd-mod-ucode=y
CONFIG_PACKAGE_snmpd=y
CONFIG_PACKAGE_softflowd=y
CONFIG_PACKAGE_ss=y
CONFIG_PACKAGE_strace=y
CONFIG_PACKAGE_strongswan=y
CONFIG_PACKAGE_strongswan-charon=y
CONFIG_PACKAGE_strongswan-libnttfft=y
CONFIG_PACKAGE_strongswan-mod-af-alg=y
CONFIG_PACKAGE_strongswan-mod-bliss=y
CONFIG_PACKAGE_strongswan-mod-chapoly=y
CONFIG_PACKAGE_strongswan-mod-curve25519=y
CONFIG_PACKAGE_strongswan-mod-hmac=y
CONFIG_PACKAGE_strongswan-mod-kernel-netlink=y
CONFIG_PACKAGE_strongswan-mod-mgf1=y
CONFIG_PACKAGE_strongswan-mod-newhope=y
CONFIG_PACKAGE_strongswan-mod-ntru=y
CONFIG_PACKAGE_strongswan-mod-openssl=y
CONFIG_PACKAGE_strongswan-mod-pem=y
CONFIG_PACKAGE_strongswan-mod-pkcs1=y
CONFIG_PACKAGE_strongswan-mod-pubkey=y
CONFIG_PACKAGE_strongswan-mod-random=y
CONFIG_PACKAGE_strongswan-mod-sha3=y
CONFIG_PACKAGE_strongswan-mod-socket-default=y
CONFIG_PACKAGE_strongswan-mod-vici=y
CONFIG_PACKAGE_strongswan-mod-x509=y
CONFIG_PACKAGE_strongswan-swanctl=y
CONFIG_PACKAGE_sysfsutils=y
CONFIG_PACKAGE_tc-bpf=y
CONFIG_PACKAGE_tcpdump=y
CONFIG_PACKAGE_terminfo=y
CONFIG_PACKAGE_ucode-mod-html=y
CONFIG_PACKAGE_ucode-mod-log=y
CONFIG_PACKAGE_ucode-mod-lua=y
CONFIG_PACKAGE_ucode-mod-math=y
CONFIG_PACKAGE_ucrun=y
CONFIG_PACKAGE_udhcprelay=y
CONFIG_PACKAGE_uhttpd=y
CONFIG_PACKAGE_uhttpd-mod-lua=y
CONFIG_PACKAGE_uhttpd-mod-ubus=y
CONFIG_PACKAGE_uhttpd-mod-ucode=y
CONFIG_PACKAGE_umbim=y
# CONFIG_PACKAGE_urandom-seed is not set
# CONFIG_PACKAGE_urngd is not set
CONFIG_PACKAGE_usbids=y
CONFIG_PACKAGE_usbutils=y
CONFIG_PACKAGE_ustp=y
CONFIG_PACKAGE_vallumd=y
CONFIG_PACKAGE_vxlan=y
CONFIG_PACKAGE_wireguard-tools=y
CONFIG_PACKAGE_wireless-regdb=y
CONFIG_PACKAGE_wwan=y
CONFIG_PACKAGE_xfrm=y
CONFIG_PACKAGE_xl2tpd=y
CONFIG_PACKAGE_zlib=y
CONFIG_PKG_ASLR_PIE_ALL=y
# CONFIG_PKG_ASLR_PIE_REGULAR is not set
CONFIG_PKG_CC_STACKPROTECTOR_ALL=y
# CONFIG_PKG_CC_STACKPROTECTOR_REGULAR is not set
# CONFIG_PKG_CHECK_FORMAT_SECURITY is not set
CONFIG_STRACE_LIBDW=y
# CONFIG_STRACE_NONE is not set
CONFIG_STRONGSWAN_ROUTING_TABLE="220"
CONFIG_STRONGSWAN_ROUTING_TABLE_PRIO="220"
CONFIG_SURICATA_BPF=y
CONFIG_SURICATA_NFQUEUE=y
# CONFIG_TARGET_EXT4_JOURNAL is not set
# CONFIG_TARGET_INITRAMFS_COMPRESSION_NONE is not set
CONFIG_TARGET_INITRAMFS_COMPRESSION_ZSTD=y
CONFIG_TARGET_KERNEL_PARTSIZE=256
CONFIG_TARGET_ROOTFS_PARTSIZE=768
CONFIG_TARGET_ROOTFS_PERSIST_VAR=y
CONFIG_TARGET_ROOTFS_SQUASHFS=y
CONFIG_TARGET_SQUASHFS_BLOCK_SIZE=256
CONFIG_TESTING_KERNEL=y
CONFIG_USE_LLVM_HOST=y
CONFIG_ZLIB_OPTIMIZE_SPEED=y
# CONFIG_KEEPALIVED_IPTABLES is not set
# CONFIG_LLDPD_WITH_CDP is not set
# CONFIG_LLDPD_WITH_EDP is not set
# CONFIG_LLDPD_WITH_FDP is not set
# CONFIG_LLDPD_WITH_SONMP is not set
# CONFIG_OPENSSL_WITH_IDEA is not set
# CONFIG_OPENSSL_WITH_MDC2 is not set
# CONFIG_OPENSSL_WITH_SEED is not set
# CONFIG_OPENSSL_WITH_WHIRLPOOL is not set
CONFIG_PACKAGE_getrandom=y
# CONFIG_PACKAGE_kmod-crypto-kpp is not set
# CONFIG_PACKAGE_libip4tc is not set
# CONFIG_PACKAGE_libip6tc is not set
# CONFIG_PACKAGE_libxtables is not set
```

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by stintel** [comment]
Looks like I attached the wrong compile.txt when reporting this issue. That is now fixed.

I have also tried bumping to 1.4.3, which exhibits the same problem. I have also tried using the OpenWrt LLVM toolchain rather than the one installed on the host, by enabling CONFIG_BPF_TOOLCHAIN_BUILD_LLVM=y, but still getting the same error.

**Comment by rockdrilla** [comment]
@stintel I've seen your [commit](https://github.com/openwrt/openwrt/commit/2d350fded62d559d5a037d0febc9c806ae4f1d6e). Does it build successfully (on qoriq)?

**Comment by stintel** [comment]
It does not. But I found some other issues, e.g. It picks up my host's bpftool, v7.5.0 while we have v7.4.0 in OpenWrt. I tried adding PKG_BUILD_DEPENDS:=bpftool/host but it still picked up the one in /usr rather than the one in staging_dir. I'll do some more digging later.

**Comment by stintel** [comment]
For reference, #16787 did not fix this, but I suspect the combination of #16787 and 5fdd18243eecf405737939bde553e14b6eec0043 does.
